mesonbuild/options.py

```diff
@@ -1412,15 +1412,17 @@ def initialize_from_subproject_call(self,
         for o in itertools.chain(project_default_options, spcall_default_options):
             keystr, valstr = o.split('=', 1)
             key = OptionKey.from_string(keystr)
-            assert key.subproject is None
-            key = key.evolve(subproject=subproject)
+            if key.subproject is None:
+                key = key.evolve(subproject=subproject)
+            elif key.subproject == subproject:
+                raise MesonException(f'subproject name not expected here: "{key}"')
```

Comment:
This gets triggered in a case like `project('foo', deafult_options : {'foo:c_args': []})`? If that's the case maybe a slightly more descriptive error here?

Reply:
I'm not sure of how to improve it... For now I left it as is but I'll take suggestions.

Reply:
Why should it be an error? I think this could simply be a warning.

To improve the error message, it could say something like "please use `c_args` instead of `foo:c_args` for project default options"

Reply:
It was silently ignored before 1.8.x so there's no reason to accept it now.

Reply:
Maybe something like 'Project cannot use it's own name as a subproject name in default_options. replace "{key}" with "{key.evolve(subproject=None)}"'

Reply:
Yes I came up with something similar after Charles's suggestion